driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/ListCollectionsPublisherImpl.java

```diff
@@ -76,6 +90,77 @@ public ListCollectionsPublisher<T> comment(@Nullable final BsonValue comment) {
 
     AsyncReadOperation<AsyncBatchCursor<T>> asAsyncReadOperation(final int initialBatchSize) {
         return getOperations().listCollections(getNamespace().getDatabaseName(), getDocumentClass(), filter, collectionNamesOnly,
-                initialBatchSize, maxTimeMS, comment);
+                authorizedCollections, initialBatchSize, maxTimeMS, comment);
+    }
+
+    <U> ListCollectionsPublisher<U> map(final Function<T, U> mapper) {
```

Comment:
Is a map method needed? Doesn't seem to be a pattern we use else where for publishers?

Reply:
### 1) I'm not 100% sure its worth adding a ListCollection Mapper type to wrap the Observable / Iterable.

If we don't have

- `ListCollectionsIterableImpl.map` returning `ListCollectionsIterable`
- `ListCollectionsPublisherImpl.map` returning `ListCollectionsPublisher`

how will

- `com.mongodb.client.internal.MongoDatabaseImpl.listCollectionNames`
- `com.mongodb.reactivestreams.client.internal.MongoDatabaseImpl.listCollectionNames`

be implemented?

### 2) As listCollectionNames is a helper built on top of listCollections it be easier if it automatically sets authorizedCollections=true.

`listCollectionNames` can set `authorizedCollections=true` automatically only if it know when a user wants `authorizedCollections=true`, and when a user doesn't want it. But how will `listCollectionNames` know that information without a user explicitly providing it?

### 3) The listCollections and listDatabases API should be inline - so either way this may require another ticket to bring the listDatabase method inline.

1. They have already been different: there is `ListDatabasesIterable.nameOnly`, while the same option for `ListCollectionsIterable` can be set only internally by the driver when `listCollectionNames` is called.
2. The `ListCollectionsPublisherImpl.map` method added in the PR is not part of API, it's an internal method. While a similar method could be added to `ListDatabasesPublisherImpl`, it will be unused. I don't think we should add unused internal method to `ListDatabasesPublisherImpl` only because a similar method is present on `ListCollectionsPublisherImpl`.